macros/src/main/scala/com/evolutiongaming/kryo/Serializer.scala

```diff
@@ -40,9 +40,32 @@ object Serializer {
 
       def isValueClass(tpe: Type) = tpe <:< typeOf[AnyVal] && tpe.typeSymbol.asClass.isDerivedValueClass
 
-      def valueClassArgType(tpe: Type) = {
-        // for value classes, first class declaration is its single field value
-        tpe.decls.head.asMethod.returnType
+      def valueClassArg(tpe: Type) = tpe.decls.head  // for value classes, first declaration is its single field value
+
+      def valueClassArgType(tpe: Type) = valueClassArg(tpe).asMethod.returnType
+
+      case class Reader(name: TermName, tree: Tree)
+      val readers = new mutable.LinkedHashMap[Type, Reader]
```

Comment:
why not TreeMap ?

Reply:
TreeMap can be an option but it is immutable and required var to update it during generation of readers and writers

Reply:
...and it can be trecky to set right ordering to avoid orders like "r1", "r11", "r2" for simplest string comparator.